File: FDWaveformView Tests/FDWaveformView_Tests.m
//
//  FDWaveformView_Tests.m
//  FDWaveformView Tests
//
//  Created by William Entriken on 5/3/14.
//  Copyright (c) 2014 William Entriken. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "FDWaveformView.h"

@interface FDWaveformView_Tests : XCTestCase
@property (nonatomic) FDWaveformView *waveformView;
@end

@implementation FDWaveformView_Tests

- (void)setUp
{
    [super setUp];
    self.waveformView = [[FDWaveformView alloc] init];
}

- (void)tearDown
{
    [super tearDown];
    self.waveformView = nil;
}

- (void)testInheritsFromUIView
{
    XCTAssert([self.waveformView isKindOfClass:[UIView class]], @"Must inherit from UIView");
}

- (void)testDelegateSetGet
{
    id <FDWaveformViewDelegate> class = nil;
    self.waveformView.delegate = class;
    XCTAssertEqual(self.waveformView.delegate, class, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testAudioURLSetGet
{
    NSURL *url = [NSURL URLWithString:@"http://www.google.com/fakesound.mp3"];
    self.waveformView.audioURL = url;
    XCTAssertEqual(self.waveformView.audioURL, url, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testTotalSamplesGet
{
    XCTAssert(self.waveformView.totalSamples == 0, @"Must have correct default value %s", __PRETTY_FUNCTION__);
}

- (void)testProgressSamplesSetGet
{
    unsigned long int samples = 0;
    self.waveformView.progressSamples = samples;
    XCTAssertEqual(self.waveformView.progressSamples, samples, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testZoomSamplesStartSetGet
{
    unsigned long int samples = 0;
    self.waveformView.zoomStartSamples = samples;
    XCTAssertEqual(self.waveformView.zoomStartSamples, samples, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testZoomSamplesEndSetGet
{
    unsigned long int samples = 0;
    self.waveformView.zoomEndSamples = samples;
    XCTAssertEqual(self.waveformView.zoomEndSamples, samples, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testDoesAllowScrubbingSetGet
{
    BOOL b = true;
    self.waveformView.doesAllowScrubbing = b;
    XCTAssertEqual(self.waveformView.doesAllowScrubbing, b, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testDoesAllowStretchAndScrollSetGet
{
    BOOL b = true;
    self.waveformView.doesAllowStretchAndScroll = b;
    XCTAssertEqual(self.waveformView.doesAllowStretchAndScroll, b, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testWavesColorSetGet
{
    UIColor *color = [UIColor purpleColor];
    self.waveformView.wavesColor = color;
    XCTAssertEqual(self.waveformView.wavesColor, color, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

- (void)testProgressColorSetGet
{
    UIColor *color = [UIColor purpleColor];
    self.waveformView.progressColor = color;
    XCTAssertEqual(self.waveformView.progressColor, color, @"Must have same value as set for %s", __PRETTY_FUNCTION__);
}

@end

File: Tests/FDWaveformViewTests.swift
//
//  FDWaveformViewTests.swift
//  FDWaveformViewTests
//
//  Created by FDWaveformView_OWNER on TODAYS_DATE.
//  Copyright  2016 FDWaveformView_OWNER. All rights reserved.
//

import XCTest
@testable import FDWaveformView

class FDWaveformViewTests: XCTestCase {

    var waveformView: FDWaveformView!

    override func setUp() {
        super.setUp()
        waveformView = FDWaveformView()
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }

    func testExample() {
        XCTAssert(true)
        // This is an example of a functional test case.
        // Use XCTAssert and related functions to verify your tests produce the correct results.
    }

    func testFD() {
        XCTAssert(waveformView != nil)
    }

    func testZoomSaples() {
        XCTAssert(waveformView.zoomStartSamples == 0)
        XCTAssert(waveformView.zoomEndSamples == 0)
    }

    func testGesturesPermissions() {
        XCTAssert(waveformView.doesAllowScroll == true)
        XCTAssert(waveformView.doesAllowStretch == true)
        XCTAssert(waveformView.doesAllowScrubbing == true)
    }

    func testColors() {
        XCTAssert(waveformView.wavesColor == UIColor.black)
        XCTAssert(waveformView.progressColor == UIColor.blue)
    }

}
